types/stateproof.go

```diff
@@ -1,8 +1,22 @@
 package types
 
+import "github.com/algorand/go-sumhash"
+
+// EncodedStateProof represents the msgpack encoded state proof.
+type EncodedStateProof []byte
+
+// MessageHash represents the message that a state proof will attest to.
+type MessageHash [32]byte
+
 // StateProofType identifies a particular configuration of state proofs.
 type StateProofType uint64
 
+const MaxHashDigestSize = sumhash.Sumhash512DigestSize
+
+// GenericDigest is a digest that implements CustomSizeDigest, and can be used as hash output.
+//msgp:allocbound GenericDigest MaxHashDigestSize
+type GenericDigest []byte
```

Comment:
Since you're adding this back in, could you update the block type to use it as well?

Reply:
I specifically removed this type at the last minute after Idan suggested I avoid including too many SP specific types. If it's being added back you can revert this change:
https://github.com/algorand/go-algorand-sdk/pull/378/files#diff-77409c5f41482f194140901cdad16d5bf6e5e6b3591f275dbf3660f12615078cL193